include/deal.II/numerics/matrix_tools.h

```diff
@@ -786,30 +786,6 @@ namespace MatrixTools
    * described in the general documentation. This function works on the
    * classes that are used to wrap PETSc objects.
    *
-   * <b>Important:</b> This function is not very efficient: it needs to
-   * alternatingly read and write into the matrix, a situation that PETSc does
-   * not handle well. In addition, we only get rid of rows corresponding to
-   * boundary nodes, but the corresponding case of deleting the respective
-   * columns (i.e. if @p eliminate_columns is @p true) is not presently
-   * implemented, and probably will never because it is too expensive without
-   * direct access to the PETSc data structures. (This leads to the situation
-   * where the action indicated by the default value of the last argument is
-   * actually not implemented; that argument has <code>true</code> as its
-   * default value to stay consistent with the other functions of same name in
-   * this namespace.)
-   *
-   * This function is used in step-17 and step-18.
-   */
-  void
-  apply_boundary_values (const std::map<types::global_dof_index,PetscScalar> &boundary_values,
```

Comment:
If we don't deprecate `PETScWrappers::SparseMatrix`, it might be a good idea to use `PETScWrappers::MatrixBase` instead of `PETScWrappers::MPI::SparseMatrix` below.